forward.c

```diff
@@ -320,14 +320,13 @@ rr_data_t forward_request(void *thread_data, rr_data_t request) {
 					if (data[1]->code < 400)
 						noauth = 1;
 					loop = 1;
-					goto shortcut;
```

Comment:
@jschwartzenberg here the goto statement can be easily removed because it only avoids executing the code that I put in an "else" block below. It was confusing because it jumps to a code that is executed when loop==0 but loop is just set to 1.

The while loop is more trickier to remove because some logic is executed during request (loop==0) other during response (loop==1) but other in both iterations. It is not so trivial to refactor it.